compiler/rustc_typeck/src/check/fn_ctxt/checks.rs

```diff
@@ -291,7 +324,51 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             } else {
                 false
             };
-            param_count_error(expected_arg_count, supplied_arg_count, "E0061", false, sugg_unit);
+
+            // are we passing elements of a tuple without the tuple parentheses?
+            let expected_arg_types = if expected_arg_tys.is_empty() {
+                // In most cases we can use expected_arg_tys, but some callers won't have the type
+                // information, in which case we fall back to the types from the input expressions.
+                fn_inputs
+            } else {
+                &expected_arg_tys
+            };
+
+            let sugg_tuple_wrap_args = if let [expected_arg_type] = expected_arg_types[..] {
+                let expected_arg_type = self.resolve_vars_if_possible(expected_arg_type);
+
+                if let ty::Tuple(expected_elems) = expected_arg_type.kind() {
+                    let expected_types: Vec<_> =
+                        expected_elems.iter().map(|k| k.expect_ty()).collect();
+                    let supplied_types: Vec<_> =
+                        args.iter().map(|arg| self.check_expr(arg)).collect();
```

Comment:
I believe `check_expr` is idempotent/the right thing to call here - let's see what the test run yields